ipalib/constants.py

```diff
@@ -340,3 +340,16 @@
 # Apache's mod_ssl SSLVerifyDepth value (Maximum depth of CA
 # Certificates in Client Certificate verification)
 MOD_SSL_VERIFY_DEPTH = '5'
+
+# subuid / subgid counts are hard-coded
+# An interval of 65536 uids/gids is required to map nobody (65534).
+SUBID_COUNT = 65536
+
+# upper half of uid_t (uint32_t)
+SUBID_RANGE_START = 2 ** 31
+# theoretical max limit is UINT32_MAX-1 ((2 ** 32) - 2)
+# We use a smaller value to keep the topmost subid interval unused.
+SUBID_RANGE_MAX = (2 ** 32) - (2 * SUBID_COUNT)
+SUBID_RANGE_SIZE = SUBID_RANGE_MAX - SUBID_RANGE_START
```

Comment:
This goes back to my previous lack of understanding of subuids. I'm asking this based on the assumption that userX gets range SUBUID_RANGE_START + SUBID_COUNT, userY gets range SUBID_RANGE_START + SUBID_COUNT + 1, etc.

So when we display range size in the idrange plugin how does SUBID_COUNT come into play?

Reply:
The ``count`` is the size of a subid range including the start value. The algorithm for automatic assignment would look like this:

```
subid_range_start = SUBID_RANGE_START + (n *  SUBID_COUNT)
n += 1
```

where ``n` is a counter starting at ``0``.


The first user owns the range ``[SUBID_RANGE_START, SUBID_RANGE_START + SUBUID_COUNT - 1]``. The second user owns the next batch of 65536 ids.

Reply:
I have added additional documentation about subid allocation.